openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2236,6 +2215,118 @@ The following is a non-normative example of a Credential Request with Credential
 
 The value of the `credential` claim in the Credential Response MUST be a string that is an SD-JWT VC. Credentials of this format are already suitable for transfer and, therefore, they need not and MUST NOT be re-encoded.
 
+# Claims Description {#claims-description}
+
+A claims description object is an object used to describe how a certain claim in
+the Credential should be displayed to the End-User. It is used in the `claims`
+or `credentialSubject` parameters in the Credential Issuer metadata defined in
+(#format-profiles).
+
+For a claims description object, the following keys defined by this
+specification MAY be used to describe the claim or claims (other keys MAY also
+be used):
+
+  * `path`: REQUIRED. Array. Claim path query as defined in (#claims-path-query)
+    to identify the claim(s) in the Credential.
+  * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the
+    Credential Issuer will always include this claim in the issued Credential.
+    If set to `false`, the claim is not included in the issued Credential if the
+    wallet did not request the inclusion of the claim, and/or if the Credential
+    Issuer chose to not include the claim. If the `mandatory` parameter is
+    omitted, the default value is `false`.
+  * `value_type`: OPTIONAL. String value determining the type of value of the
+    claim. Valid values defined by this specification are `string`, `number`,
+    and image media types such as `image/jpeg` as defined in IANA media type
+    registry for images
+    (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    Other values MAY also be used.
+  * `display`: OPTIONAL. Array of objects, where each object contains display
+    properties of a certain claim in the Credential for a certain language.
+    Below is a non-exhaustive list of valid parameters that MAY be included:
+     * `name`: OPTIONAL. String value of a display name for the claim.
+     * `locale`: OPTIONAL. String value that identifies language of this object
+       represented as language tag values defined in BCP47 [@!RFC5646]. There
+       MUST be only one object for each language identifier.
+
+The order of claims description objects in the `claims` or `credentialSubject`
+array is used by the Wallet to determine the order in which the claims are
+displayed to the End-User, unless another mechanism is defined by the profile.
+
+# Claims Path Query {#claims-path-query}
+
+The metadata specifications above make use of claims path queries to select a
+particular claim in the credential or a set of such claims. This appendix
+defines the syntax and semantics of claims path queries.
+
+A claims path query MUST be a non-empty array of strings, `null` values, or
```

Comment:
It would be nice to support negative integers as well since those are used by CWT/COSE claims.

Reply:
For addressing claims identified by numbers in CWT/COSE, I suggest to use numbers in strings to distinguish them from array elements. WDYT?

Reply:
We don't yet have any Credential Format Profiles where claims identified by numbers in CWT/COSE.  But it might be good to have an approach for them anyway.  Using numbers in strings to distinguish them from array elements would work but then there's a potential (though unlikely) for ambiguity of a claim identified by the number vs. identified by the number as a string. Maybe the rules here could cover that rare case by just saying to select for both in the case that the credential format allows for it. 

Or use an object to represent a number claim name/key something like `{"claim-identified-by-number": -257}`

Reply:
Another approach would be use numbers similar to strings as representing the literal key value to select for. And then a to use single element or empty arrays to convey selecting on array element(s). So something like `[1]` would be use where just `1` is being used now. And an empty `[]` array (or some variation) could be used in place of how `null` is being used now.

Something like this, for example: 

 `["degrees", [], "type"]`: All `type` claims in the `degrees` array are selected.
 `["nationalities", [1]]`: The second nationality is selected.

Reply:
Another approach would be to use a different property instead of `path`. Perhaps the simplest option would be to use something like `path_cbor` that could then contain negative integer values?

Reply:
I like this proposal, in particular using `[]` instead of `null` as a wildcard

I don't know it there might be issues with some implementations, as the ones mentioned for `""`. I can say that in my implementations I can handle void instanced of arrays or strings without any problem.

Reply:
Using an Array would also allow you to
- select everything via `[]` 
- select a specific entry `[1]`
- select several sepcific entries (which other approaches can't easyil do `[0,1]`

That sounds like the cleanest option to me so far imho.

Reply:
We only have `null` in DCQL in OpenID4VP for now. I suggest that we extend this as need as soon as we have a concrete use case.